### PR TITLE
Add tests to persistence component

### PR DESCRIPTION
**Pull request by chamilaadhi**
- Add tests to persistence component
- Refactor RegistryPersistanceUtil
- Remove classes without any usages

Unit test for https://github.com/wso2/product-apim/issues/9370

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/test/java/org/wso2/carbon/apimgt/persistence/mapper/APIMapperTestCase.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.persistence/src/test/java/org/wso2/carbon/apimgt/persistence/mapper/APIMapperTestCase.java:2]: Need to update the license header

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/test/java/org/wso2/carbon/apimgt/persistence/utils/RegistryPersistenceUtilTestCase.java [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+    @Test
+    public void testTenantDomain() {
+        PowerMockito.mockStatic(ServerConfiguration.class);
+        ServerConfiguration config = Mockito.mock(ServerConfiguration.class);
+        PowerMockito.when(ServerConfiguration.getInstance()).thenReturn(config);
+        PowerMockito.when(config.getFirstProperty(anyString())).thenReturn(null);
+      
+        String domain = RegistryPersistenceUtil
+                .getTenantDomain(new APIIdentifier("test@" + TENANT_DOMAIN, "test", "1.0"));
+        //Assert.assertEquals("Tenant domain does not match", TENANT_DOMAIN, domain);
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.persistence/src/test/java/org/wso2/carbon/apimgt/persistence/utils/RegistryPersistenceUtilTestCase.java:233]: Commented code

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/test/java/org/wso2/carbon/apimgt/persistence/mapper/APIMapperTestCase.java (hunk 1 above)
Comment (chamilaadhi) [reply components/apimgt/org.wso2.carbon.apimgt.persistence/src/test/java/org/wso2/carbon/apimgt/persistence/mapper/APIMapperTestCase.java]: done

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/test/java/org/wso2/carbon/apimgt/persistence/utils/RegistryPersistenceUtilTestCase.java (hunk 2 above)
Comment (chamilaadhi) [reply components/apimgt/org.wso2.carbon.apimgt.persistence/src/test/java/org/wso2/carbon/apimgt/persistence/utils/RegistryPersistenceUtilTestCase.java]: removed

**Review by mushthaq33 — APPROVED** [review]

**Pull request merged**